### PR TITLE
Pass bearer token

### PR DESCRIPTION
**Pull request by boryashkin**
#123 

An example for the issue

### CODE REVIEW

**Comment by boryashkin** [comment]
I found no way to add a hook to constructors of existing providers, so it will be only possible with the CustomProvider (user of the lib could easily copy/paste params of preexisting providers and construct it via AddCustomProvider or NewCustom)

File: provider/oauth2.go [hunk 1]
```diff
@@ -217,6 +221,11 @@ func (p Oauth2Handler) AuthHandler(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	if p.bearerTokenHook != nil {
```
[the remark this answers is not in the document]
Comment (boryashkin) [reply provider/oauth2.go]: done, i thought to add a conditional logging, like "nil token" and "passed a token successfuly", but discarded the idea

File: provider/oauth2.go [hunk 2]
```diff
@@ -56,6 +57,9 @@ func (u UserData) Value(key string) string {
 	return ""
 }
 
+// BearerTokenHook accepts provider name, user and token, received during oauth2 authentication
+type BearerTokenHook func(string, token.User, oauth2.Token)
```
[the remark this answers is not in the document]
Comment (boryashkin) [reply provider/oauth2.go]: done

**Review by umputun — APPROVED** [review]
LGTM, thx

**Pull request merged**